packages/firestore/test/unit/specs/describe_spec.ts

```diff
@@ -242,7 +244,64 @@ export function describeSpec(
   } else {
     specsInThisTest = {};
     builder();
-    const output = JSON.stringify(specsInThisTest, null, 2);
+    // Note: We use json-stable-stringify instead of JSON.stringify() to ensure
+    // that the generated JSON does not produce diffs merely due to the order
+    // of the keys in an object changing.
+    const output = stringify(specsInThisTest, { space: 2, cmp: stringifyCmp });
     writeJSONFile(output);
   }
 }
+
+/**
+ * The key ordering overrides used when sorting keys in the generated JSON.
+ * If both keys being compared are present in this array then they should be
+ * ordered in the generated JSON in the same relative order of this array.
+ */
+const stringifyCustomOrdering = [
+  'describeName',
+  'itName',
+  'tags',
+  'config',
+  'steps'
+];
+
+/**
+ * A comparator function for stringify() that sorts keys in the JSON output.
+ *
+ * In order to produce semi-readable JSON that has an intuitive top-level key
+ * arrangement, some keys are sorted non-alphabetically; namely, the ordering
+ * defined in `stringifyCustomOrdering` is used, falling back to alphabetical
+ * ordering if one or both of the key are not overridden.
+ */
+function stringifyCmp(a: stringify.Element, b: stringify.Element): number {
+  // If the keys are equal, then avoid the costly computations below.
```

Comment:
Performance doesn't matter here so I'd optimize for readability over anything else.

You can make this simpler by creating a function that determines the group number (use e.g. `stringifyCustomOrdering.length` for `expected` and `stringifyCustomOrdering.length + 1` for "everything else").

Then you can just do:

```ts
const aGroup = stringifyGroup(a.key);
const bGroup = stringifyGroup(b.key);

if (aGroup === bGroup) {
  return primitiveComparator(a.key, b.key);  // from util/misc.ts
} else {
  return primitiveComparator(aGroup, bGroup);
}
```

Reply:
Done.  Great suggestion.